src/logits_processor.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include <cstdint>
+#include <cstdio>
+#include <fstream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <sys/types.h>
+
+#include "generators.h"
+#if USE_GUIDANCE
+#include "llguidance.h"
+#endif
+
+// #if USE_CUDA
+// #include "cuda/cuda_common.h"
+// #include "models/kernels.h"
+// #endif
+
+#include "logits_processor.h"
+
+namespace Generators {
+
+#if USE_GUIDANCE
+GuidanceLogitsProcessor::GuidanceLogitsProcessor(const State& state)
+    : vocab_size_(state.params_->config.model.vocab_size),
+      eos_token_(state.params_->config.model.eos_token_id),
+      batch_size_(state.params_->search.batch_size),
+      device_type_(state.params_->p_device->GetType()),
+      params_(state.params_),
+      guidance_type_(state.params_->guidance_type),
+      guidance_data_(state.params_->guidance_data) {
```

Comment:
Since we hold a pointer to the params, can we avoid making a copy for all the parameters here and just use the params to get them as needed?

Reply:
Just keeping eos_token_ as there a type change to uint.